### DISCUSSION TITLE
promise/avoid-new: do we really want to enforce node's util.promisify in the browser?

### DISCUSSION THREAD

**Discussion by darkbasic** [standard/eslint-config-standard-with-typescript#1950] [General]
util.promisify belongs to node.js, do we really want to enforce it for browsers as well?

**Comment by mightyiam** [comment]
Thank you for bringing this up, @darkbasic.

I know it is provided by Node.js and not available in browsers. But many general JS APIs are not available in browsers. That's how we build for the browser, right? We add whatever seems useful.

`util.promisify` seems like a generally useful JS utility.

Is it difficult to include in the browser?

**Reply by darkbasic** [reply]
It's not "difficult", but increases the bundle size and not everybody might be ok with that.

**Reply by mightyiam** [reply]
At risk of stating the obvious, everybody being okay with a rule was never a criteria.

**Reply by darkbasic** [reply]
At risk of stating the obvious if a linting rule forces you to install a ~200 stars 2 years old dead npm package (https://github.com/browserify/node-util) the linting rule might be at fault.
Apart from being dead and unmaintained the package has many issues: https://github.com/browserify/node-util/pull/62

**Reply by mightyiam** [reply]
What about [pify](https://www.npmjs.com/package/pify)?

**Reply by darkbasic** [reply]
Better, but still more than two years without a single commit/release.

**Reply by mightyiam** [reply]
Possibly because it's "done"?

**Reply by darkbasic** [reply]
I didn't run an npm audit but I'm always skeptical of packages that don't update dependencies on a regular basis. Chances are one of its dependencies might contain some sort of security vulnerability.

**Reply by mightyiam** [reply]
For what it's worth, it has only dev deps.

**Reply by darkbasic** [reply]
Good, that makes it a suitable alternative.
